ecdsa/src/recovery.rs

```diff
@@ -285,24 +289,36 @@ where
     /// Recover a [`VerifyingKey`] from the given `prehash` of a message, the
     /// signature over that prehashed message, and a [`RecoveryId`].
     ///
-    /// This function has been modified to support SP1 acceleration for secp256k1 signature verification
-    /// in the context of a zkVM. If the curve is secp256k1 and the prehash is 32 bytes long (indicating a SHA-256 hash),
-    /// the function will use [`crate::sp1::VerifyingKey::recover_from_prehash_secp256k1`] to return a precomputed public key.
+    /// This function has been modified to support SP1 acceleration for secp256k1/r1 signature verification
+    /// in the context of a zkVM. If the curve is secp256k1 or secp256r1 and the prehash is 32 bytes long (indicating a SHA-256 hash),
+    /// the function will use [`crate::sp1::VerifyingKey::recover_from_prehash_secp256`] to return a precomputed public key.
     #[allow(non_snake_case)]
     pub fn recover_from_prehash(
         prehash: &[u8],
         signature: &Signature<C>,
         recovery_id: RecoveryId,
     ) -> Result<Self> {
-        // Only override the recovery function if this is a secp256k1 curve and the prehash is 32 bytes long (indicating a SHA-256 hash).
+        // Only override the recovery function if this is a secp256k1 or secp256r1 curve and the prehash is 32 bytes long (indicating a SHA-256 hash).
         cfg_if::cfg_if! {
             if #[cfg(all(target_os = "zkvm", target_vendor = "succinct"))] {
                 // Reference: https://en.bitcoin.it/wiki/Secp256k1.
                 const SECP256K1_ORDER: [u8; 32] = hex_literal::hex!("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141");
+                const SECP256R1_ORDER: [u8; 32] = hex_literal::hex!("FFFFFFFF00000000FFFFFFFFFFFFFFFFBCE6FAADA7179E84F3B9CAC2FC632551");
+
+                let curve = if C::ORDER == <C as Curve>::Uint::decode_field_bytes(GenericArray::from_slice(&SECP256K1_ORDER)) {
+                    Secp256Curve::K1
+                } else if C::ORDER == <C as Curve>::Uint::decode_field_bytes(GenericArray::from_slice(&SECP256R1_ORDER)) {
+                    Secp256Curve::R1
+                } else {
+                    return Err(Error::new()); // Unsupported curve
+                };
```

Comment:
I believe there are other curves that use `ecdsa-core` that are not `Secp256r1` or `Secp256k1`. 

If the curve is not `Secp256r1` or `Secp256k1`, you should no-op, similar to what it did before.